### PR TITLE
Front-end integration completed

### PR DESCRIPTION
**Pull request by yinbogao**
Hi Justin,

For the past two weeks we've been working on error handling, testing, as well as a new front-end design. The new frontend html was based on an open-sourced template by Ajlkn. Free for personal and commercial use under the CCA 3.0 license. 

## New Front page with colors picked from colormind.io
snip20180430_98

## User-submit library search completed
snip20180430_99

## Error handling: Return error if snippet is not in mei format.
snip20180430_100

## Error handling: Return error message if not found and start new search from the same page.
snip20180430_101

## New Search Results display with alternating color
snip20180430_102

### CODE REVIEW

**Comment by justinnhli** [comment]
Can I get a link to the hosted version?

**Comment by justinnhli** [comment]
Also, the linting needs to be fixed

File: search.py [hunk 1]
```diff
@@ -94,6 +98,8 @@ def get_title(path):
     """
     tree, _ = prepare_tree(path)
     title_stmt = get_elements(tree, 'titleStmt')
+    if not title_stmt:
+        return "Title Not Found"
```
Comment (justinnhli) [inline search.py]: A better design is to return None, then display "Title Not Found" on the front end.

**Comment by Lijia-Li** [comment]
Link: https://project-rechord.herokuapp.com/

I frankly don’t know how to fix the lint there. QAQ

File: search.py [hunk 2]
```diff
@@ -106,6 +112,8 @@ def get_creator(path):
     Return: creators_list [List<element>]: List of elements marking the creator(s) of a piece
     """
     tree, _ = prepare_tree(path)
+    if not get_elements(tree, 'respStmt'):
+        return "Creator Not Found"
```
Comment (justinnhli) [inline search.py]: Ditto with the `get_title()` comment.

File: ReChord_frontend.py [hunk 3]
```diff
@@ -92,14 +95,27 @@ def search_snippet(path, snippet):
     Return: rendered result page 'ReChord_result.html'
     """
     xml = BytesIO(snippet.encode())
```
Comment (justinnhli) [inline ReChord_frontend.py:97]: It's better to explicitly declare `error_msg` to be `None`, instead of leaving it undefined if there are no errors.

File: ReChord_frontend.py [hunk 4]
```diff
@@ -92,14 +95,27 @@ def search_snippet(path, snippet):
...
+            return  render_template('ReChord_result.html', nomatch=not_found)
+    except:
+        if etree.XMLSyntaxError or ValueError:
+            error_msg = "Invalid MEI snippet inputs. Please double check the source and try it again!"
+        elif KeyError:
+            error_msg = "Invalid upload file. Please double check the source and try it again!"
+        else:
+            error_msg = "There are some errors in the search. Please check the source."
+
+        return render_template('ReChord_result.html', errors=error_msg)
```
Comment (justinnhli) [inline ReChord_frontend.py]: The "No exception type(s) specified" lint message means you need to specify the exception. Try:

```python
except (etree.XMLSyntaxError, ValueError):
    error_msg = "Invalid MEI snippet inputs. Please double check the source and try it again!"
except KeyError:
    error_msg = "Invalid upload file. Please double check the source and try it again!"
return render_template('ReChord_result.html', errors=error_msg)
```

File: ReChord_frontend.py [hunk 5]
```diff
@@ -110,15 +126,18 @@ def search_terms(path, tag, para):
...
     results = text_box_search_folder(path, tag, para)
 
-    return render_template('ReChord_result.html', origins=results)
-
+    total_appearance = len(results)
+    if total_appearance != 0:
+        return render_template('ReChord_result.html', origins=results)
+    else:
+        not_found = "No matched term found, maybe try something else?"
+        return render_template('ReChord_result.html', nomatch=not_found)
```
Comment (justinnhli) [inline ReChord_frontend.py]: Code looks a little nicer if the condition is positive. You also have an extra variable. You can rewrite this as:

```python
if results:
    return render_template('ReChord_result.html', origins=results)
else:
    not_found = "No matched term found, maybe try something else?"
    return render_template('ReChord_result.html', nomatch=not_found)
```

File: templates/generic.html [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+			<script src="assets/js/jquery.min.js"></script>
+			<script src="assets/js/jquery.scrollex.min.js"></script>
+			<script src="assets/js/jquery.scrolly.min.js"></script>
+			<script src="assets/js/skel.min.js"></script>
+			<script src="assets/js/util.js"></script>
+			<!--[if lte IE 8]><script src="assets/js/ie/respond.min.js"></script><![endif]-->
+			<script src="assets/js/main.js"></script>
+
+	</body>
+</html>
```
Comment (justinnhli) [inline templates/generic.html]: Why is this file here? This looks like an example, which you should delete.

File: ReChord_frontend.py [hunk 7]
```diff
@@ -40,34 +40,37 @@ def my_form_post():
     Return: rendered result page 'ReChord_result.html' by call on helper functions
     """
```
Comment (justinnhli) [inline ReChord_frontend.py:42]: I'm okay with you ignoring the pylint warning:

`# pylint: disable=(too-many-return-statements`

File: templates/index.html [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+                        </div>
+                        <input type="file" name="base_file" multiple="" required>
+                        <input type="submit" name="submit" value="Upload and Search Parameter">
+                    </form>
+                </section>
+            </div>
+        </div>
+    </section>
+
+    <!-- Two -->
```
Comment (justinnhli) [inline templates/index.html]: What's this stuff here?

Comment (yinbogao) [reply templates/index.html]: Ughhh that was the placeholder I typed, will delete.

File: templates/generic.html (hunk 6 above)
Comment (yinbogao) [reply templates/generic.html]: On it

File: ReChord_frontend.py (hunk 5 above)
Comment (Lijia-Li) [reply ReChord_frontend.py]: deleted the extra variable. What do you mean by positive condition?

File: templates/generic.html (hunk 6 above)
Comment (yinbogao) [reply templates/generic.html]: template file deleted

File: search.py [hunk 9]
```diff
@@ -113,7 +113,7 @@ def get_creator(path):
     """
     tree, _ = prepare_tree(path)
     if not get_elements(tree, 'respStmt'):
-        return "Creator Not Found"
+        return None
```
Comment (justinnhli) [inline search.py]: Is this being handled correctly on the front end?

**Comment by Lijia-Li** [comment]
@justinnhli Done!

File: templates/index.html (hunk 8 above)
Comment (justinnhli) [reply templates/index.html]: @yinbogao @Lijia-Li delete?

Comment (Lijia-Li) [reply templates/index.html]: done! I think we will add in introduction and user documentation to the page at some point tonight. @justinnhli

**Comment by justinnhli** [comment]
looks good to me. Not sure if you're adding documentation, so let me know when it's ready to be merged

**Comment by Lijia-Li** [comment]
Could you merge please? Bo will do another PR later.
